EIPS/eip-XXXX.md

```diff
@@ -0,0 +1,151 @@
+- It must include a custom header `X-EIP-XXXX-ABI-ENCODED-INPUT` that suggests the call signature the user must invoke at `X-EIP-XXXX-CONTRACT-ADDRESS` to gain access to the endpoint.
+
+### Example Response
+
+```
+HTTP/1.1 402 Payment Required
+Date: Mon, 01 Aug 2022 13:37:00 GMT
+Content-Type: application/json
+X-EIP-XXXX-CONTRACT-ADDRESS: 0x005241438cAF3eaCb05bB6543151f7AF894C5B58
+X-ABI-ENCODED-INPUT: 23b872dd00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000007b
+
+"{\"constant\":false,\"inputs\":[{\"name\":\"_from\",\"type\":\"address\"},{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"_value\",\"type\":\"uint256\"}],\"name\":\"transferFrom\",\"outputs\":[{\"name\":\"success\",\"type\":\"bool\"}],\"type\":\"function\"}"
+```
+
+### Authorization via EIP-712 Signature
+
+After successfully calling the suggested Ethereum contract with the appropriate inputs, for a user request that containing a validly signed message for endpoint authorization, the server must allow access to the bespoke endpoint.
+
+The signature is complaint with [EIP-712](./eip-712.md):
+
+```js
+keccak256(abi.encodePacked(
+  hex"1901",
+  DOMAIN_SEPARATOR,
+  keccak256(abi.encode(
+    keccak256("Invocation(address contract,string input"),
+    contract,
+		keccak256(bytes(input)),
+  ))
+))
+```
+
+where `DOMAIN_SEPARATOR` must be unique to the chain to prevent replay attacks from other domains, and satisfy the requirements of [EIP-712](./eip-712.md):
+
+```js
+DOMAIN_SEPARATOR = keccak256(
+	abi.encode(
+		keccak256(
+			"EIP712Domain(string version,uint256 chainId,string endpoint,string method)"
+		),
+		keccak256(bytes(version)),
+		chainid,
+		keccak256(bytes(endpoint)),
+		keccak256(bytes(method))
+	)
+);
+```
+
+where `endpoint` must represent the full URL, e.g. `https://ethereum.org/api/v1/cutedoge` and `method` an existing HTTP method.
+
+```js
+{
+  "types": {
+    "EIP712Domain": [
+      {
+        "name": "version",
+        "type": "string"
+      },
+      {
+        "name": "chainId",
+        "type": "uint256"
+      },
+      {
+        "name": "endpoint",
+        "type": "string"
+      },
+      {
+        "name": "method",
+        "type": "string"
+      }
+    ],
+    "Invocation": [
+      {
+        "name": "contract",
+        "type": "address"
+      },
+      {
+        "name": "input",
+        "type": "string"
+      }
+    ],
+    "primaryType": "Invocation",
+    "domain": {
+      "version": version,
+      "chainId": chainid,
+      "endpoint": endpoint,
+      "method": method
+    },
+    "message": {
+      "contract": contract,
+      "input": input
+    }
+  }
+}
+```
+
+The resulting hexadecimal-encoded signature must be included in a user's request to the endpoint as the `Bearer` value of the `Authorization` header. An example:
```

Comment:
I would propose creating a new authentication scheme for this, e.g. `Signed 0xabc`.